### ISSUE TITLE
typescript not found

### ISSUE THREAD

**Issue by guybedford** [frankwallis/plugin-typescript#9]
I'm installing the plugin into a custom name - `jspm install ts=github:frankwallis/plugin-typescript@master`.

As a result, something, somewhere is expecting to be able to load `typescript` without having it declared as a dependency in its own execution context.

Generally all requires should be declared as dependencies so that naming is portable like this.

It seems like this is either in TypeScript itself on compile or the `incremental-compiler.js` file as it gets to the `translating` step before giving the 404 error in the logging.

**Comment by guybedford** [comment]
Actually I do have typescript installed as a global too, and am still getting this.

The actual error is GET file:///Users/guybedford/Projects/jspm-cli/sandbox/typescript/bin/lib.d.ts net::ERR_FILE_NOT_FOUND.

**Comment by frankwallis** [comment]
This is because when compiling it needs to fetch this declaration file which is shipped with the typescript compiler. I need to make a change so that it resolves it correctly, but until then you have to add an explicit mapping to the file - see https://github.com/frankwallis/plugin-typescript/issues/6

My mapping looks like this:
```
"typescript/bin/*.d.ts": "jspm_packages/npm/typescript@1.4.1/bin/*.d.ts",
```

**Comment by guybedford** [comment]
You can call `System.normalize('typescript', __moduleName).then(function(normalized){ return System.locate({ name: normalized, metadata: {} }); }).then(function(pathToTypeScript) { })` to access these paths through the loader API.

**Comment by frankwallis** [comment]
thanks, that's now fixed in 0.5.7.

**Comment by frankwallis** [comment]
Actually there is still an issue here - if the root project does not have typescript installed then the resolution fails.

It seems to be because __moduleName is set to ```http://localhost:8081/jspm_packages/github/frankwallis/plugin-typescript@0.5.10/lib/format-errors.js``` which is actually the module address. I've traced it through to here:
```
// Define our eval outside of the scope of any other reference defined in this
// file to avoid adding those references to the evaluation scope.
function __eval(__source, __global, load) {
  // Hijack System.register to set declare function
  var __curRegister = System.register;
  System.register = function(name, deps, declare) {
    if (typeof name != 'string') {
      declare = deps;
      deps = name;
    }
    // store the registered declaration as load.declare
    // store the deps as load.deps
    load.declare = declare;
    load.depsList = deps;
  }
  try {
    eval('(function() { var __moduleName = "' + (load.name || '').replace('"', '\"') + '"; ' + __source + ' \n }).call(__global);');
  }
  catch(e) {
    if (e.name == 'SyntaxError' || e.name == 'TypeError')
      e.message = 'Evaluating ' + (load.name || load.address) + '\n\t' + e.message;
    throw e;
  }
```

when it does the eval, load.name is correctly set to ```github:frankwallis/plugin-typescript@0.5.10```, but __source is set to this:

```
"var __moduleAddress = "http://localhost:8081/jspm_packages/github/frankwallis/plugin-typescript@0.5.10/lib/format-errors.js";System.register("http://localhost:8081/jspm_packages/github/frankwallis/plugin-typescript@0.5.10/lib/format-errors.js", ["typescript"], function($__export) {
  "use strict";
  var __moduleName = "http://localhost:8081/jspm_packages/github/frankwallis/plugin-typescript@0.5.10/lib/format-errors.js";
  var ts;
  function formatErrors(diags, logger) {
    /* MORE CODE */
```

so the good value of __moduleName gets overwritten by a bad value coming from traceur by the look of it.

**Comment by guybedford** [comment]
@frankwallis thanks I've created https://github.com/ModuleLoader/es6-module-loader/issues/332.

**Comment by guybedford** [comment]
@frankwallis what version of Traceur are you using here? We've definitely got this in the tests, so I'm not sure why it's failing for you (`System.traceurOptions.moduleName = false` should make sure Traceur doesn't set this).

**Comment by guybedford** [comment]
@frankwallis is this specifically for builds?

**Comment by frankwallis** [comment]
@guybedford - I'll have to get back to you later about this - what do you mean by 'specifically for builds'?

**Comment by guybedford** [comment]
Sure thanks, no rush. I meant as in if the error came up while running builds or in the browser. Because the in-browser case is very much tested - https://github.com/ModuleLoader/es6-module-loader/blob/master/test/loader/moduleName.js, also why I suggested ensuring Traceur is up to date.

**Comment by frankwallis** [comment]
It is happening in the browser, when compiling the plugin-typescript example project. What I am seeing is that with traceur 0.0.84 (default version from jspm) it fails as described above (even if I set moduleName: false), but when I manually update traceur to 0.0.87 then everything works as expected.

**Comment by guybedford** [comment]
@frankwallis ok thanks. It sounds like this is due to that previous Traceur regression then, which should fix up in the next update.

**Comment by frankwallis** [comment]
this was resolved in jspm 0.15
